### PR TITLE
Add @Priority and make it the default

### PR DESCRIPTION
**Pull request by rbygrave**

### CODE REVIEW

**Comment by norrisjeremy** [comment]
I like this, as it allows using Priority based injection without having to pull in potentially problematic javax.annotation-api dependency.

File: inject-test/src/test/java/org/example/coffee/priority/base/CBasei.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import javax.inject.Singleton;
+
+@Singleton
+@Priority(1)
+public class CBasei implements BaseIface {
+
+  @Override
+  public String other() {
+    return "a";
```
Comment (norrisjeremy) [inline inject-test/src/test/java/org/example/coffee/priority/base/CBasei.java:13]: Minor nit, but should this be `return "c";`?

File: inject-test/src/test/java/io/avaje/inject/SystemContextTest.java [hunk 2]
```diff
@@ -16,28 +17,30 @@
   @Test
   public void getBeansByPriority() {
 
-    final List<Somei> beans = SystemContext.getBeansByPriority(Somei.class);
-    assertThat(beans).hasSize(2);
+    final List<Somei> beans = SystemContext.context().getBeansByPriority(Somei.class, Priority.class);
```
[the remark this answers is not in the document]
Comment (rbygrave) [reply inject-test/src/test/java/io/avaje/inject/SystemContextTest.java:20]: Yes, I'll merge and sort that out - thanks.

File: inject-test/src/test/java/org/example/coffee/priority/base/CBasei.java (hunk 1 above)
Comment (rbygrave) [reply inject-test/src/test/java/org/example/coffee/priority/base/CBasei.java:13]: Yes it should, it will be nice to fix it - thanks

**Pull request merged**